core/os/memory.cpp

```diff
@@ -120,38 +109,31 @@ void *Memory::realloc_static(void *p_memory, size_t p_bytes, bool p_pad_align) {
 
 	if (prepad) {
 		mem -= PAD_ALIGN;
-		uint64_t *s = (uint64_t *)mem;
 
 #ifdef DEBUG_ENABLED
-		if (p_bytes > *s) {
-			atomic_add(&mem_usage, p_bytes - *s);
+		uint64_t *mem64 = (uint64_t *)mem;
+
+		if (p_bytes > *mem64) {
+			atomic_add(&mem_usage, p_bytes - *mem64);
 			atomic_exchange_if_greater(&max_usage, mem_usage);
 		} else {
-			atomic_sub(&mem_usage, *s - p_bytes);
+			atomic_sub(&mem_usage, *mem64 - p_bytes);
 		}
 #endif
 
 		if (p_bytes == 0) {
 			free(mem);
 			return NULL;
 		} else {
-			*s = p_bytes;
-
 			mem = (uint8_t *)realloc(mem, p_bytes + PAD_ALIGN);
 			ERR_FAIL_COND_V(!mem, NULL);
 
-			s = (uint64_t *)mem;
-
-			*s = p_bytes;
```

Comment:
Not completely sure, but this looks to be required for the mem_usage tracking

Reply:
Your absolutely right, I've just changed it.